packages/@storefront/core/src/services/url.ts

```diff
@@ -157,15 +157,38 @@ class UrlService extends BaseService<UrlService.Options> {
 
   filterState(state: Store.State) {
     const { session: { config, ...session }, data, ...rootConfig } = state;
-    const products = this.app.config.history.length === 0 ? [] : data.present.products;
+    let { navigations, products, template } = data.present;
+    let {
+      navigations: autocompleteNavigations,
+      products: autocompleteProducts,
+      template: autocompleteTemplate,
+    } = data.present.autocomplete;
+
+    if (this.app.config.history.length === 0) {
+      autocompleteNavigations = [];
+      autocompleteProducts = products = [];
```

Comment:
I know we do this in the tests, but afaik we general don't in the code https://github.com/airbnb/javascript#variables--no-chain-assignment

Reply:
Sounds good, will remove.